New sample: Display device location with autopan modes (#571)
Esri/arcgis-runtime-samples-java: map_view/display-device-location-with-autopan-modes/README.md

# Display device location with autopan modes

Display your position on the map and switch between different types of autopan modes.

> ![Image of display device location with autopan modes](DisplayDeviceLocationWithAutopanModes.png)

Reviewer: I think a wee gif here would look great: the static image doesn't show a user much about what the sample does other than its UI set up.

## Use case

When using a map within a GIS, it may be helpful for a user to know their own location within a map, whether that's to aid the user's navigation or to provide an easy means of identifying/collecting geospatial information at their location.

## How to use the sample

> Click the checkbox to show the location symbol at a simulated location on the map. Select one of the autopan modes from the drop down box:

Reviewer: The sample starts with the checkbox already selected, we can reorder this a little: 

```suggestion
The sample loads with a symbol marking a simulated location on the map. Explore the available autopan modes by selecting an option from the drop down box :
```

* Off - Shows the location with no autopan mode set.

> * Re-Center - In this mode, the map re-centers on the location symbol when the symbol moves outside a "wander extent".

Reviewer: let's rename this in the drop down to match this notation (currently the drop down is "recenter")

* Navigation -  This mode is best suited for in-vehicle navigation.
* Compass - This mode is better suited for waypoint navigation when the user is walking.

> 

Reviewer: ```suggestion
Uncheck the box to stop the location display. 
```

## How it works

1. Create a `MapView`.

> 2. Get the `LocationDisplay` object by calling `getLocationDisplay()` on the `MapView`.

Reviewer: since we've already mentioned the class map view above, we can refer to it as plain english afterwards: 
```suggestion
2. Get the `LocationDisplay` object by calling `getLocationDisplay()` on the map view.
```

2. Create a `SimulatedLocationDataSource` and call its `setLocations()` method, passing the route `Polyline` and new `SimulationParameters` as parameters. 

> 3. Start the `LocationDisplay` with `startAsync()` to begin receiving location updates.

Reviewer: ```suggestion
3. Start the location display with `startAsync()` to begin receiving location updates.
```

> 5. Use `locationDisplay.setAutoPanMode` to change how the map behaves when location updates are received.

Reviewer: this behaviour is set on the map view, not the map: let's update this text to reflect that
```suggestion
5. Use `locationDisplay.setAutoPanMode()` to change how the map view behaves when location updates are received.
```

## Relevant API

> * ArcGISMap

Reviewer: I'd argue this isn't a relevant API for this sample: ArcGISMap is relevant in ALL 2D samples, but unless it's forming the main part of a sample (like "Display a Map, set zoom scale, set location etc) we don't need to include it.

* LocationDisplay
* LocationDisplay.AutoPanMode
* MapView
* SimulatedLocationDataSource
* SimulationParameters

## Additional information

A custom set of points (provided in JSON format) is used to create a `Polyline` and configure a `SimulatedLocationDataSource`. The simulated location data source enables easier testing and allows the sample to be used on devices without an actively updating GPS signal. To display a user's real position, use `NMEALocationDataSource` instead.

## Tags

> compass, GPS, location, map, mobile, navigation

Reviewer: the location display is on the map view not the map, I wonder if map view is better as a tag here. But probably best to go back to the design author for this, and it shouldn't block this sample being merged.
